PublishedModules/Psychokiller1888/Calculator/Calculator.py

```diff
@@ -0,0 +1,83 @@
+import math
+
+from core.base.model.Intent import Intent
+from core.base.model.Module import Module
+from core.dialog.model.DialogSession import DialogSession
+
+
+class Calculator(Module):
+	"""
+	Author: Psychokiller1888
+	Description: Do some calculation with alice
+	"""
+
+	_INTENT_MATHS = Intent('Maths')
+
+
+	def __init__(self):
+		self._SUPPORTED_INTENTS = [
+			self._INTENT_MATHS
+		]
+		self._lastNumber = None
+		super().__init__(self._SUPPORTED_INTENTS)
+
+
+	def onMessage(self, intent: str, session: DialogSession) -> bool:
+		if not self.filterIntent(intent, session):
+			return False
+
+		slots = session.slotsAsObjects
+		sessionId = session.sessionId
+
+		if intent == self._INTENT_MATHS:
+			if ('Left' not in slots and 'Right' not in slots) or 'Function' not in slots:
+				self.continueDialog(sessionId=sessionId, text=self.TalkManager.randomTalk('notUnderstood'))
+				return True
+
+			func = slots['Function'][0].value['value']
+
+			if 'Left' in slots and 'Right' in slots:
+				result = self.calculate(float(slots['Left'][0].value['value']), float(slots['Right'][0].value['value']), func)
+
+			elif 'Right' in slots and 'Left' not in slots:
+				if not self._lastNumber:
```

Comment:
when the last result was 0 it will say that there was no previous result. Maybe check for Null instead